docs/addons.md

```diff
@@ -0,0 +1,167 @@
+
+**[istio][]**: Adds the core [Istio][istio-docs] services.
+
+**[jaeger][]**: Deploy the [Jaeger Operator][jaeger-docs] in the “simplest” configuration.
+
+**[knative][]**: Adds the [Knative][knative-docs] middleware to your cluster.
+
+**[linkerd][]**: Deploys the linkerd service mesh. [See below](#linkerd) for configuration.
+
+**[metrics-server][]**: Adds the Kubernetes Metrics Server for API access to service metrics.
+
+**[prometheus]()**: Deploys the [Prometheus Operator][prometheus-docs].
+
+**[rbac]()**: Enable Role Based Access Control for authorisation. Note that this is incompatible with some other add-ons ([See notes](#rbac))
+
+**[registry][]**: Deploy a private image registry and expose it on localhost:32000. The storage add-on will be enabled as part of this add-on. See the registry documentation for more details.
+
+**[storage][]**: Create a default storage class which allocates storage from a host directory.
+
+
+### dns
+
+As noted above, this deploys CoreDNS to supply address resolution services to Kubernetes. This service is commonly required by other addons, so it is recommended that you enable it.
+
+By default it points to Google's 8.8.8.8, 8.8.4.4 servers for resolving
+addresses. This can be changed by running the command:
+
+```bash
+microk8s.kubectl -n kube-system edit configmap/coredns
+```
+
+This will invoke the `vim` editor so you can alter the configuration.
+
+### dashboard
+
+The standard Kubernetes Dashboard is a convenient way to keep track of the
+activity and resource use of MicroK8s
+
+
+To log in to the Dashboard, you will need the access token. This is generated randomly on deployment, so a few commands are needed to retrieve it:
+
+```bash
+token=$(microk8s.kubectl -n kube-system get secret | grep default-token | cut -d " " -f1)
+microk8s.kubectl -n kube-system describe secret $token
+```
+
+Next, you need a connection to the API server. While the MicroK8s snap will have an IP address on your local network, the recommended way to do this is through the proxy service. You can initiate the proxy with the command:
+
+```bash
+microk8s.kubectl proxy --accept-hosts=.* --address=0.0.0.0 &
+```
+
+You can then access the Dashboard at the address
+
+[http://127.0.0.1:8001/api/v1/namespaces/kube-system/services/https:kubernetes-dashboard:/proxy/]()
+
+![IMAGE of Dashboard]()
+
+
+### fluentd
+
+
+### gpu
+
+This addon enables GPU support for MicroK8s. Note that this is obviously dependent on the host system having suitable Nvidia GPU hardware and
+relevant drivers.
+
+With the GPU addon enabled, workloads can request the GPU using a limit setting, `nvidia.com/gpu: 1 `. For example:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: cuda-vector-add
+spec:
+  restartPolicy: OnFailure
+  containers:
+    - name: cuda-vector-add
+      image: "k8s.gcr.io/cuda-vector-add:v0.1"
+      resources:
+        limits:
+          nvidia.com/gpu: 1
+```
+
```

Comment:
I feel there may be some troubleshooting connected to this - I guess we can add a link here